File: backend/providers/gemini.js
module.exports = {
  
  name: 'gemini',
  aliases: ['google'],
  displayName: 'Google AI Studio',
  description: 'Google AI Studio — Gemini models via OpenAI-compatible endpoint',
  baseUrl: 'https://generativelanguage.googleapis.com/v1beta/openai/',
  apiMode: 'openai_chat',
  envVars: ['GEMINI_API_KEY', 'GEMINI_BASE_URL'],
  authType: 'api_key',
  defaultModel: 'gemini-2.0-flash',
  defaultMaxTokens: 8192,
  signupUrl: 'https://aistudio.google.com',
  supportsHealthCheck: true,
  modelProfiles: {
    'gemini-2': { supportsReasoning: true, supportsThinking: false, combinedBudget: false, contextWindow: 1048576, maxOutputTokens: 8192 },
    'gemini-1': { supportsReasoning: false, supportsThinking: false, combinedBudget: false, contextWindow: 1048576, maxOutputTokens: 8192 },
    '*': { supportsReasoning: false, supportsThinking: false, combinedBudget: false, contextWindow: 1048576, maxOutputTokens: 4096 },
  },

};

File: backend-py/app/providers/gemini.py
"""
Provider config for Google AI Studio.
"""

from __future__ import annotations

from typing import Optional

INFO = {
    "name": "Google AI Studio",
    "default_model": "gemini-2.0-flash",
    "default_max_tokens": 8192,
    "api_mode": "openai_chat",
}


def resolve_base_url() -> Optional[str]:
    return "https://generativelanguage.googleapis.com/v1beta/openai"


def resolve_api_key(env_key: Optional[str] = None) -> Optional[str]:
    return env_key
